File: CS/Shared/Northwind/NorthwindContext.cs
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DevExpress.CRUD.Northwind {
    public class NorthwindContext : DbContext {
        static NorthwindContext() {
            Database.SetInitializer(new NorthwindContextInitializer());
        }
        protected override void OnModelCreating(DbModelBuilder modelBuilder) {
            modelBuilder.Entity<Product>().Property(x => x.Name).IsRequired();
        }
        public DbSet<Category> Categories { get; set; }
        public DbSet<Product> Products { get; set; }
    }
}

File: VB/Shared/Northwind - VB/NorthwindContext.vb
Imports System
Imports System.Collections.Generic
Imports System.Data.Entity
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace DevExpress.CRUD.Northwind
	Public Class NorthwindContext
		Inherits DbContext

		Shared Sub New()
			Database.SetInitializer(New NorthwindContextInitializer())
		End Sub
		Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
			modelBuilder.Entity(Of Product)().Property(Function(x) x.Name).IsRequired()
		End Sub
		Public Property Categories() As DbSet(Of Category)
		Public Property Products() As DbSet(Of Product)
	End Class
End Namespace
